crates/uv-install-wheel/src/wheel.rs

```diff
@@ -431,22 +431,41 @@ fn install_script(
         Err(err) => return Err(Error::Io(err)),
     }
     let size_and_encoded_hash = if start == placeholder_python {
-        let is_gui = {
-            let mut buf = vec![0; 1];
-            script.read_exact(&mut buf)?;
-            if buf == b"w" {
-                true
-            } else {
-                script.seek_relative(-1)?;
-                false
+        // Read the rest of the first line, one byte at a time, until we hit a newline.
```

Comment:
It's pretty rare for there to be any trailing content here, so I think reading one-byte-at-a-time is probably fine. But we could also read (e.g.) 256 bytes into the buffer if desired.